cloud_function_go/function.go

```diff
@@ -0,0 +1,137 @@
+// Package p contains an HTTP Cloud Function.
+package p
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+func shouldRetry(statusCode int) bool {
+	retry := true
+	switch statusCode {
+	case http.StatusBadRequest:
+		fmt.Printf("Got HTTP %d bad request, skip retry\n", statusCode)
+		retry = false
+	case http.StatusNotFound:
+		fmt.Printf("Got HTTP %d not found, skip retry\n", statusCode)
+		retry = false
+	case http.StatusUnauthorized:
+		fmt.Printf("Got HTTP %d unauthorized, skip retry\n", statusCode)
```

Comment:
If you get unauthorized its mean the token is incorrect . Let put more informative message